test/python/transpiler/test_basic_swap.py

```diff
@@ -368,6 +368,51 @@ def test_far_swap_with_gate_the_middle(self):
         after = pass_.run(dag)
 
         self.assertEqual(circuit_to_dag(expected), after)
+      
+    
+    def test_fake_run(self):
+        """A fake run, doesn't change dag
+        q0:--(+)-------.--
+              |        |
+        q1:---|--------|--
+              |
+        q2:---|--------|--
+              |        |
+        q3:---.--[H]--(+)-
+
+        CouplingMap map: [0]--[1]--[2]--[3]
+
+        q0:-------(+)-------.---
+                   |        |
+        q1:-----X--.--[H]--(+)--
+                |
+        q2:--X--X---------------
+             |
+        q3:--X------------------
+
+        """
+        coupling = CouplingMap([[0, 1], [1, 2], [2, 3]])
+
+        qr = QuantumRegister(4, "q")
+        circuit = QuantumCircuit(qr)
+        circuit.cx(qr[3], qr[0])
+        circuit.h(qr[3])
+        circuit.cx(qr[0], qr[3])
+        dag = circuit_to_dag(circuit)
+
+        # expected iff fake_run=False
+        expected = QuantumCircuit(qr)
+        expected.swap(qr[3], qr[2])
+        expected.swap(qr[2], qr[1])
+        expected.cx(qr[1], qr[0])
+        expected.h(qr[1])
+        expected.cx(qr[0], qr[1])
+
+        pass_ = BasicSwap(coupling, fake_run=True)
+        after = pass_.run(dag)
+
+        self.assertNotEqual(circuit_to_dag(expected), after)
+        self.assertEqual(dag, after)
```

Comment:
This test mostly looks good to me, thanks.  One small addition, just to check that `fake_run` is functioning correctly: could we modify it slightly so we check the output `property_set` for the `"final_layout"` key, and check it comes out the same between the `fake_run=True` and `fake_run=False` cases?

It'd use the same circuit as your test, but with the assertions looking something along the lines of:
```python
circuit = ...

fake_pm = PassManager([BasicSwap(coupling, fake_run=True)])
real_pm = PassManager([BasicSwap(coupling, fake_run=False)])

self.assertEqual(circuit, fake_pm.run(circuit))
self.assertNotEqual(circuit, real_pm.run(circuit))
self.assertIsInstance(fake_pm.property_set["final_layout"], Layout)
self.assertEqual(fake_pm.property_set["final_layout"], real_pm.property_set["final_layout"])
```
(I may have made some mistakes in that.)

It tests the same "fake run doesn't crash" and "fake run doesn't modify the input" properties as yours, but also ensures that the logical 'output' of the fake run is emitted too.